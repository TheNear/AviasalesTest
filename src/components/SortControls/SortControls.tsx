import React from "react";
import style from "./SortControls.module.css";
import Button from "../../componentsHelper/Button/Button";
import { useQuery } from "../../hooks/useQuery";

const SortControls: React.FC = () => {
  const query = useQuery();

  const sortByDuration = () => {
    query.queryPush("sort", "by_time");
  };

  const sortByPrice = () => {
    query.queryPush("sort", "by_price");
  };

  return (
    <div className={style.sort}>
      <Button text="Самый дешевый" active clickHandler={sortByPrice} />
      <Button text="Самый быстрый" clickHandler={sortByDuration} />
    </div>
  );
};

export default SortControls;
