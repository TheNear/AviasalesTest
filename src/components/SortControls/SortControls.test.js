import React from "react";
import { shallow } from "enzyme";
import SortControls from "./SortControls";

describe("Test SortControl component", () => {
  it("Render empty SortControls", () => {
    const wrapper = shallow(<SortControls />);
    expect(wrapper).toMatchSnapshot();
  });
});
