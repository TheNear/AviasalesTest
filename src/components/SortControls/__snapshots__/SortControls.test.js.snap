// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SortControl component Render empty SortControls 1`] = `
<div
  className="sort"
>
  <Button
    active={true}
    text="Самый дешевый"
  />
  <Button
    text="Самый быстрый"
  />
</div>
`;
